### ISSUE TITLE
Ensure that the option to setup Google Analytics is disabled if Analytics is disabled via googlesitekit_available_modules

### ISSUE THREAD

**Issue by jamesozzie** [google/site-kit-wp#5071]
## Bug Description

If Google Analytics has been disabled via _**googlesitekit_available_modules**_ (https://github.com/google/site-kit-wp/issues/3993) ensure that the option to setup Analytics isn't included for other admins who decide to set up the plugin. 

![image](https://user-images.githubusercontent.com/41326532/162797869-6f959adf-ef5b-45cd-9d56-90019e7070a3.png)

https://user-images.githubusercontent.com/41326532/162797693-acce4a38-0451-48a3-be7c-a4c84fcff6d3.mp4

## Steps to reproduce

1. Disable Analytics via the below filter
2. Install and setup SK
3. The option to add GA remains

Note that GA is never added to the setup steps, so the setup flow does complete successfully if a user proceeds. Google Analytics isn't connected at any stage, and the option to connect doesn't appear, nor for the setup CTAs. 

```
add_filter(
	'googlesitekit_available_modules',
	function( $modules ) {
		return array( 'search-console' );
	}
);
```

## Additional Context

 - PHP Version: 
 - Browser: [e.g. chrome, safari]
 - Plugin Version: [e.g. 22]

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by tofumatt** [comment]
Looks like we thought ahead with our API and have a separate return value for `isModuleActive`: https://github.com/google/site-kit-wp/blob/dce817b95df607384f4ed5e039bd6836e28d9475/assets/js/googlesitekit/modules/datastore/modules.js#L773-L775

We should use the `null` return value to mean the module isn't available—I tested this out by using the filter above and `useSelect( ( select ) => select( CORE_MODULES ).isModuleConnected( 'analytics' ) );` returned `null` as-expected, so we'll want to use that `null` return value to change output/prevent selectors from running.

I think the components are the wrong place to prevent the selectors' resolvers from running—the resolvers themselves should check to see if the module is enabled and resolve immediately (with "empty" values) if the module isn't enabled. That's a cleaner and more-scalable solution… either one is a fair bit of work but I think doing it in the datastore is better.

For components I tried checking for `null` instead of false-y `isModuleConnected` and it worked great… here's a proof-of-concept PR for the screen used in the issue summary: #5120

**Comment by tofumatt** [comment]
I think the ACs here might've lead me to think we want to really protect against any Analytics requests and build in a robust protection against running code when a module is disabled, but I don't think that's needed.

Maybe I misunderstood the intent of the ACs, but they seemed kind of broad in scope to me. I've written up an IB to fix this issue and ones like it throughout the codebase for Analytics, but I think it can be discussed a bit before moving this to execution 🙂

**Comment by eugene-manuilov** [comment]
> If we adjust components to not output Analytics component when the module is not available everything should work fine. Here's a list of places we're checking for a false-y `isModuleConnected` where we should explicitly check for `=== false` instead:

@tofumatt, yes, I think it should help but I would defer it to @felixarntz to make the final decision.

Just want to add that Analytics widgets won't be rendered anyway because the `whenActive` hook won't render them, so we don't need to update selectors there. What we need to update are widgets in AdSense and Search Console modules as well as for WP dashboard and admin bar to make sure that they get Analytics reports only if the Analytics module is active.

Plus, we need to update the `SetupUsingProxyWithSignIn` component not to render the `ActivateAnalyticsNotice` element if the Analytics module is not available.

**Comment by felixarntz** [comment]
@tofumatt Definitely +1 to making sure that both `isModuleActive` and `isModuleConnected` return `null` in case the module doesn't exist (and `false` only if it exists but the conditions aren't met). I think this is an enhancement worth making regardless of the rest of the approach.

Still, I think the IB is a bit overly simplistic, as in some areas we rely on `isModuleActive( 'analytics' )` and we can't just replace that with `isModuleConnected( 'analytics' )` in every case. For example, on the splash screen the CTA is explicitly only relevant if Analytics isn't active, but not if it's active but not connected.

Also, +1 to @eugene-manuilov's comment that we shouldn't need to update most Analytics widgets if they already have the `whenActive` wrapper.

Overall, while per my first paragraph I support the idea, I consider the differentiation between `null` and `false` a rather error-prone, almost "hidden", functionality that may be easily missed also in the future. Someone reading `isModuleConnected === false` in the future may just randomly replace it with `! isModuleConnected`, thinking that it is the same. That's why I'd be in favor of adding an additional check, maybe via a new boolean selector `hasModule( moduleSlug )` on the `core/modules` store.

**Comment by tofumatt** [comment]
> Overall, while per my first paragraph I support the idea, I consider the differentiation between `null` and `false` a rather error-prone, almost "hidden", functionality that may be easily missed also in the future. Someone reading `isModuleConnected === false` in the future may just randomly replace it with `! isModuleConnected`, thinking that it is the same. That's why I'd be in favor of adding an additional check, maybe via a new boolean selector `hasModule( moduleSlug )` on the `core/modules` store.

That's fair, and something we already occasionally encounter with our "`undefined`-as-loading/not-yet-loaded" approach to selector return values. I actually wanted to highlight that here because while it might be tempting to say "Oh, but we already do this", I think in retrospect that's not even the best solution and one we should be careful to protect against in the future. (If I could, I'd change all of the loading values to be a magic constant or something we could better check against! 😅 )

A `hasModule`-type selector (that ultimately relies on the existing `null` return value) seems like a good plan—it's more explicit but still not a lot of effort. I'll update the IB with that approach 👍🏻

**Comment by felixarntz** [comment]
@tofumatt IB mostly looks good, a few small notes:
> * Add a new selector to the `CORE_MODULES` data store: `isModuleAvailable( slug )`. It should accept a module slug as the sole argument, and return:
>  * `undefined` if `isModuleConnected( slug ) === undefined`
>  * `false` if `isModuleConnected( slug ) === null`  
>  * `true` if neither of the above conditions are met

It strikes me a bit odd to rely on `isModuleConnected` here. This is simply about whether the module exists, and while I'm all for reusing code, I think using `getModule` here instead better represents its purpose. `isModuleConnected` also checks something else in addition to whether the module exists, we essentially don't care about its `true` vs `false` state. Hence relying on `getModule` here is sufficient and a bit closer to the purpose.

> * Adjust components to not output Analytics component when the module is not available. Here's a list of places we're checking for a false-y `isModuleConnected`

I think we may want to also reference any false-y `isModuleActive` selector calls here, since that can also apply in a few places. Like anywhere that we render some sort of CTA for Analytics when it's not active, we should also not render such a thing, i.e. add the `isModuleAvailable` check to the condition.

**Comment by FlicHollis** [comment]
Hey @tofumatt just following up on this one as this looks like one that's nearly ready + it will unblock / inform 5072 + 5074.

**Comment by felixarntz** [comment]
IB ✅

**Comment by mohitwp** [comment]
### QA Update ✅ 

- Verified on dev and latest environment.
- I'm able to reproduce issue on latest environment.
- Verified following steps mentioned under QAB.
- Verified that the Analytics CTA doesn't show on the splash screen.
- Verified that user is able to access site kit dashboard successfully.
- Verified that successful message for analytics set up is not showing.
- Verified that analytics and optimize module removed successfully.
- Verified that Site Kit not raising errors due to the Analytics module being programmatically removed as an available module.
- Verified that main, entity dashboard, WP dashboard and WP menu bar not showing any error. 
- Verified tag manager is connected successfully, not showing any error and working. 
- Verified GTM snippet in page source.

<details>

https://user-images.githubusercontent.com/94359491/193585244-0f6cd10f-d8e6-4b1e-b106-c5eabcbd4d8c.mp4

![image](https://user-images.githubusercontent.com/94359491/193587719-3101bb64-0be6-44ba-a263-622fd28f57b7.png)

</details>
